Python/syntax_test_python_strings.py

```diff
@@ -445,3 +430,120 @@
 #             ^^ constant.other.placeholder.python
 #               ^ - constant.other.placeholder.python
 #                ^^ constant.other.placeholder.python
+
+datetime.datetime.utcnow().strftime("%Y%m%d%H%M")
+#                                    ^^^^^^^^^^ constant.other.placeholder
+
+"My String %% %s"
+#          ^^ constant.other.placeholder
+#             ^^ constant.other.placeholder
+
+"My String %(s)s %s"
+#          ^^^^^ constant.other.placeholder
+#            ^ variable.other.placeholder
+#                ^^ constant.other.placeholder
+
+"Testing {:,.2f}".format(1000)
+#        ^^^^^^^ constant.other.placeholder
+#        ^ punctuation.definition.placeholder.begin
+#              ^ punctuation.definition.placeholder.end
+
+"Testing {0:>9,}".format(1000)
+#        ^^^^^^^ constant.other.placeholder
+
+"Testing {:j^9,}".format(1000)
+#        ^^^^^^^ constant.other.placeholder
+#         ^^^^^ constant.other.format-spec
+
+"result: {value:{width}.{precision}}"
+#        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ constant.other.placeholder
+#              ^^^^^^^^^^^^^^^^^^^^ meta.format-spec.python
+#        ^ punctuation.definition.placeholder.begin
+#               ^^^^^^^ constant.other.placeholder constant.other.placeholder
+#               ^ punctuation.definition.placeholder.begin
+#                       ^^^^^^^^^^^ constant.other.placeholder constant.other.placeholder
+#                                 ^^ punctuation.definition.placeholder.end
+
+f"string"
+# <- storage.type.string
+#^^^^^^^^ string.quoted.double
+
+ RF"""string"""
+#^^ storage.type.string
+#^^^^^^^^^^^^^^ meta.string.interpolated string.quoted.double.block
+
+F'''string'''
+# <- storage.type.string
+#^^^^^^^^^^^^ meta.string.interpolated string.quoted.single.block
+
+ rf'string'
+#^^ storage.type.string
+#^^^^^^^^^^ meta.string.interpolated string.quoted.single
+
+
+# FIXME The opening brace incorrectly gets its scope stack reset
+# due to https://github.com/SublimeTextIssues/Core/issues/1526.
+# Thus, the following tests on it are failing.
+
+f"{something}"
+#^^^^^^^^^^^^ meta.string.interpolated
+# <- storage.type.string
+#^ punctuation.definition.string.begin
+# ^ punctuation.definition.interpolation.begin
```

Comment:
Well, this one is difficult since you cannot split interpolated "sections" over lines in single-line strings but you can split multi-line strings. Unfortunately, the syntax definition doesn't differ between these cases yet. I'll take a look soon.

I wonder whether I should use `section` and `definition` for them or just `section`, though.